src/posix/platform/settings.cpp

```diff
@@ -32,429 +32,442 @@
  *
  */
 
+#include "platform-posix.h"
+
 #include <assert.h>
+#include <inttypes.h>
 #include <stddef.h>
-#include <stdlib.h>
-#include "utils/wrap_string.h"
-
-#include <openthread-core-config.h>
+#include <sys/stat.h>
 
+#include <openthread/platform/misc.h>
 #include <openthread/platform/settings.h>
 
-#include <code_utils.h>
+#include "common/code_utils.hpp"
 
-#include "flash.h"
+#define VerifyOrDie(aCondition) \
+    do                          \
+    {                           \
+        if (!(aCondition))      \
+        {                       \
+            perror(__func__);   \
+            exit(EXIT_FAILURE); \
+        }                       \
+    } while (false)
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+static const size_t kMaxFileNameSize = sizeof(OPENTHREAD_CONFIG_POSIX_SETTINGS_PATH) + 32;
 
-enum
-{
-    kBlockAddBeginFlag    = 0x1,
-    kBlockAddCompleteFlag = 0x02,
-    kBlockDeleteFlag      = 0x04,
-    kBlockIndex0Flag      = 0x08,
-};
+static int sFd = -1;
```

Comment:
How about `sFd` -> `sSettingFd` and swap file `fd` -> `swapFd`? (making the names easier to distinguish when reading the code).